File: android/src/main/java/com/getcapacitor/community/audio/queue/QueueTrack.kt
package com.getcapacitor.community.audio.queue

import org.json.JSONObject

class QueueTrack(
    val id: String,
    val url: String,
    val name: String,
    val isMusic: Boolean,
    val forcePlay: Boolean
) {
    val assetId: String
        get() {
            return url
        }

    constructor(jsObject: JSONObject) : this(
        id = kotlin.run {
            val key = "id"
            if (jsObject.has(key)) {
                return@run jsObject.getString(key) ?: ""
            }
            return@run ""
        },
        url = kotlin.run {
            val key = "url"
            if (jsObject.has(key)) {
                return@run jsObject.getString(key) ?: ""
            }
            return@run ""
        },
        name = kotlin.run {
            val key = "name"
            if (jsObject.has(key)) {
                return@run jsObject.getString(key) ?: ""
            }
            return@run ""
        },
        isMusic = kotlin.run {
            val key = "isMusic"
            if (jsObject.has(key)) {
                return@run jsObject.getBoolean(key)
            }
            return@run false
        },
        forcePlay = kotlin.run {
            val key = "forcePlay"
            if (jsObject.has(key)) {
                return@run jsObject.getBoolean(key)
            }
            return@run false
        },
    )
}

File: ios/Plugin/Queue/QueueTrack.swift
//
//  QueueTrack.swift
//  Capacitor
//
//  Created by Олег  Руссу on 09/10/2023.
//

import Foundation


public struct QueueTrack {

    public let id: String
    public let url: String
    public let name: String
    public let isMusic: Bool
    public let forcePlay: Bool

    public var assetId: String {
        get {
            url
        }
    }

    public init(jsObject: [String: Any]) {
        self.id  = String(jsObject["id"] as? Int ?? 0)
        self.url = jsObject["url"] as? String ?? ""
        self.name = jsObject["name"] as? String ?? ""
        self.isMusic = (jsObject["isMusic"] as? Int ?? 0) == 1
        self.forcePlay = (jsObject["forcePlay"] as? Int ?? 0) == 1
    }
}
